edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs

```diff
@@ -31,8 +35,46 @@ public class Hosting
             bool clientCertAuthEnabled,
             SslProtocols sslProtocols)
         {
+            /* Define the HttpsConnectionAdapterOptions which integrates with aspnetcore https middleware to help us manage the client cert chain.
+               We need to store the client cert chain in the connection context, so that we can perform custom CA validation later to authenticate with the service identity.
+               We are using logic here that depends on an undocumented api, however eliminating this dependency would require a large refactor.
+               Context in this issue: https://github.com/dotnet/aspnetcore/issues/21606 */
+            HttpsConnectionAdapterOptions connectionAdapterOptions = new HttpsConnectionAdapterOptions()
+            {
+                ServerCertificate = serverCertificate,
+                OnAuthenticate = (context, options) =>
+                {
+                    if (clientCertAuthEnabled)
+                    {
+                        /* Certificate authorization can be enabled in edgeHub, however direct methods will hit this logic and not need certificates.
+                           This implies that the ClientCertificateRequired should be false.
+                           However, the naming of this field is not semantically correct.
+                           We looked at how aspnetcore used this field, and found that it is true when CertificateMode.AllowCertificate is set (which does not require certificates).
+                           Therefore we can set this field to true and it will still work with direct methods.
+                           */
+                        options.ClientCertificateRequired = true;
+                        options.RemoteCertificateValidationCallback = (_, clientCert, chain, policyErrors) =>
```

Comment:
Maybe out of scope for this PR, but I wonder if it is possible to do all validation inside this callback instead of going through the hassle of saving the cert chain, which seems against what aspnet intended

Reply:
Yeah we can do this, but its a large refactor. If this is working I don't see much benefit into handling this proactively. If aspnet breaks this later I think we should reconsider.